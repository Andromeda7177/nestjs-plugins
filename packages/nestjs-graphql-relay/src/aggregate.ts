import { Field, ObjectType, Int } from "type-graphql";

@ObjectType({ isAbstract: true })
export abstract class Aggregate {
  @Field(() => Int)
  readonly count: number;
}
