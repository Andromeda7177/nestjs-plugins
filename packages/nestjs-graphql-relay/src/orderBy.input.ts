import { registerEnumType, InputType, Field } from "@nestjs/graphql";

export enum OrderByDirection {
  ASC = "ASC",
  DESC = "DESC",
}

registerEnumType(OrderByDirection, { name: "OrderByDirection" });

@InputType()
export class OrderByInput {
  @Field(() => OrderByDirection, { nullable: true })
  createdAt?: OrderByDirection;

  @Field(() => OrderByDirection, { nullable: true })
  updatedAt?: OrderByDirection;
}
