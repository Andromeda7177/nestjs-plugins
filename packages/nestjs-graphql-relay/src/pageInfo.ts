import { ObjectType, Field } from "@nestjs/graphql";
import * as Relay from "graphql-relay";

@ObjectType()
export class PageInfo implements Relay.PageInfo {
  @Field(() => Boolean, { nullable: true })
  readonly hasNextPage?: boolean | null;
  @Field(() => Boolean, { nullable: true })
  readonly hasPreviousPage?: boolean | null;
  @Field(() => String, { nullable: true })
  readonly startCursor?: Relay.ConnectionCursor | null;
  @Field(() => String, { nullable: true })
  readonly endCursor?: Relay.ConnectionCursor | null;
}
