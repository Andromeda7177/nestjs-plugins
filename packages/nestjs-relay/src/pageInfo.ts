import { ObjectType, Field } from "type-graphql";
import * as Relay from "graphql-relay";

@ObjectType()
export class PageInfo implements Relay.PageInfo {
  @Field(() => Boolean, { nullable: true })
  hasNextPage?: boolean | null;
  @Field(() => Boolean, { nullable: true })
  hasPreviousPage?: boolean | null;
  @Field(() => String, { nullable: true })
  startCursor?: Relay.ConnectionCursor | null;
  @Field(() => String, { nullable: true })
  endCursor?: Relay.ConnectionCursor | null;
}
