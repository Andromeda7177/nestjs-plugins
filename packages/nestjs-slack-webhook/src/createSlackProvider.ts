import { Provider } from "@nestjs/common";
import { IncomingWebhook } from "@slack/webhook";
import { SLACK_TOKEN } from "./slackConstants";
import { getSlackClient } from "./getSlackClient";
import { SlackOptions } from "./slackOptions";

export function createSlackProvider(
  options: SlackOptions
): Provider<IncomingWebhook> {
  return {
    provide: SLACK_TOKEN,
    useValue: getSlackClient(options),
  };
}
