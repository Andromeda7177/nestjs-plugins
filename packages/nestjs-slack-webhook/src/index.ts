export * from "./injectSlack";
export * from "./slackModule";
export * from "./slackOptions";
