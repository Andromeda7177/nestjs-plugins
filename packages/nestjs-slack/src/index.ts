export * from "./slackModule";
export * from "./slackOptions";
export * from "./injectSlack";
