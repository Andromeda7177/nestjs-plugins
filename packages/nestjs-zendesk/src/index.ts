export * from "./zendesk.constants";
export * from "./zendesk.inject";
export * from "./zendesk.interface";
export * from "./zendesk.module";
