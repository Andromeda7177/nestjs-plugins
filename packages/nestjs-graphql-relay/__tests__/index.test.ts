import {
  Column,
  Connection,
  createConnection,
  Entity,
  getRepository,
  PrimaryGeneratedColumn,
} from "typeorm";
import { Factory } from "typeorm-factory";
import { findAndPaginate, getPagingParameters } from "../src";

@Entity()
class Example {
  @PrimaryGeneratedColumn()
  readonly id: number;

  @Column()
  readonly name: string;
}

describe("app", () => {
  let connection: Connection;
  beforeAll(async () => {
    connection = await createConnection({
      type: "sqlite",
      database: "nestjs-plugins",
      synchronize: true,
      entities: [Example],
    });

    const repo = getRepository(Example);
    await connection.query(`DELETE from ${repo.metadata.tableName}`);
  });

  afterAll(() => connection.close());

  it("getPagingParameters", () => {
    expect(getPagingParameters({})).toMatchInlineSnapshot(`Object {}`);
    expect(getPagingParameters({ first: 1 })).toMatchInlineSnapshot(`
      Object {
        "limit": 1,
        "offset": 0,
      }
    `);
    expect(() => getPagingParameters({ first: 1, after: "after" }))
      .toThrowErrorMatchingInlineSnapshot(`"invalid before query"`);
    expect(() => getPagingParameters({ last: 1, before: "before" }))
      .toThrowErrorMatchingInlineSnapshot(`"invalid before query"`);
  });

  describe("findAndPaginate", () => {
    it("empty", async () => {
      const res = await findAndPaginate(
        {
          where: { name: "undefined" },
        },
        {},
        getRepository(Example),
      );
      expect(res).toMatchInlineSnapshot(`
        Object {
          "edges": Array [],
          "pageInfo": Object {
            "endCursor": null,
            "hasNextPage": false,
            "hasPreviousPage": false,
            "startCursor": null,
          },
        }
      `);
    });

    it("find", async () => {
      const name = "name";
      const f = new Factory(Example).attr("name", "random name");
      await f.createList(123);
      await f.createList(123, {
        name,
      });

      const res = await findAndPaginate(
        {
          where: { name },
        },
        {},
        getRepository(Example),
      );
      res.edges.map(({ node }) => expect(node.name).toEqual(name));
      expect(res.pageInfo).toMatchSnapshot();
    });
  });
});
