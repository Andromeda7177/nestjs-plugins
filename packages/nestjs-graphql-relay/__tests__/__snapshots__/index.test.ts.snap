// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app findAndPaginate find 1`] = `
Object {
  "endCursor": "YXJyYXljb25uZWN0aW9uOjA=",
  "hasNextPage": false,
  "hasPreviousPage": false,
  "startCursor": "YXJyYXljb25uZWN0aW9uOjA=",
}
`;
